### PR TITLE
Add NSF grant EAR-1948603 to list of funding support

### LINKED ISSUE #860: Add more PyGMT-related grants
**Description of the desired feature**

In the [`README.rst`](https://github.com/GenericMappingTools/pygmt/blob/master/README.rst) file, we mention:

> The development of PyGMT was funded by NSF grant OCE-1558403.

Compared to the [GMT support page](https://www.generic-mapping-tools.org/about/#support):

- NSF grant OCE-1558403 is not mentioned in the GMT support page
- Some recent grants in the GMT support page are not listed in the PyGMT project.

Questions for @PaulWessel:

1. Should "NSF grant OCE-1558403" be added to the GMT support page? Are there any other grants that are not listed?
2. Are there any grants you think we should list in the PyGMT project?

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Add NSF grant [EAR-1948603](https://www.nsf.gov/awardsearch/showAward?AWD_ID=1948602) - 'A Sustainable Plan for the Future of the Generic Mapping Tools'. Aka the grant supporting @meghanrjones's postdoc (among other things) :smile:

Fixes #860

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by seisman — CHANGES_REQUESTED** [review]
Also need to update `doc/overview.rst`.

File: README.rst [hunk 1]
```diff
@@ -188,8 +188,9 @@ the **BSD 3-clause License**. A copy of this license is provided in
 Support
 -------
 
-The development of PyGMT was funded by
-`NSF grant OCE-1558403 <https://www.nsf.gov/awardsearch/showAward?AWD_ID=1558403>`__.
+The development of PyGMT was supported by NSF grants
```
Comment (seisman) [inline README.rst:191]: Should we change `was` to `was/is`?

Comment (weiji14) [reply README.rst:191]: Yeah I thought about that. Probably should.

Comment (seisman) [reply README.rst:191]: @PaulWessel sorry to bother you with this tiny linguistic issue.

Comment (meghanrjones) [reply README.rst:191]: I think you're looking for "has been" :)

Comment (weiji14) [reply README.rst:191]: Present perfect, thanks! Don't tell my English teacher.

**Review by seisman — APPROVED** [review]

**Pull request merged**